### PR TITLE
Add config check for invalid duplicate locale domains

### PR DESCRIPTION
**Pull request by ijjk**
This ensures we properly show an error when users attempt configuring multiple domain items with the same `defaultLocale`. We already have an existing check ensuring multiple domain items don't share the same items in their `locale` array so this continues on that check. 

This isn't currently supported as we expect to redirect to one specific domain when a locale is detected as the preferred and if there are multiple we don't know which to redirect to. 

Closes: https://github.com/vercel/next.js/pull/22337

## Bug

- [x] Related issues linked using `fixes #number`
- [x] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

### CODE REVIEW

File: packages/next/server/config.ts [hunk 1]
```diff
@@ -340,6 +340,19 @@ function assignDefaults(userConfig: { [key: string]: any }) {
...
         if (!item.domain || typeof item.domain !== 'string') return true
 
+        const defaultLocaleDuplicate = i18n.domains?.find(
+          (altItem) =>
+            altItem.defaultLocale === item.defaultLocale &&
+            altItem.domain !== item.domain
+        )
+
+        if (defaultLocaleDuplicate) {
+          console.warn(
```
Comment (styfle) [inline packages/next/server/config.ts:350]: Should this be `console.error()` instead?

**Review by styfle — APPROVED** [review]

File: packages/next/server/config.ts (hunk 1 above)
Comment (ijjk) [reply packages/next/server/config.ts:350]: We mainly use `console.warn` in the checks so went with that but an error is thrown here after this log https://github.com/vercel/next.js/blob/9cbc4aacd50d0e79be7b8178eec9219457514da9/packages/next/server/config.ts#L366

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| buildDuration | 16.6s | 16.3s | -344ms |
| buildDurationCached | 3.8s | 3.7s | -128ms |
| nodeModulesSize | 49.5 MB | 49.5 MB | ⚠️ +1.45 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.933 | 2.961 | ⚠️ +0.03 |
| / avg req/sec | 852.48 | 844.24 | ⚠️ -8.24 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.024 | 1.987 | -0.04 |
| /error-in-render avg req/sec | 1235.42 | 1258.17 | +22.75 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| 359.HASH.js gzip | 2.96 kB | 2.96 kB | ✓ |
| 745.HASH.js gzip | 180 B | 180 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 21 kB | 21 kB | ✓ |
| webpack-HASH.js gzip | 1.53 kB | 1.53 kB | ✓ |
| Overall change | 67.9 kB | 67.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| _app-HASH.js gzip | 803 B | 803 B | ✓ |
| _error-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| amp-HASH.js gzip | 554 B | 554 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.52 kB | 2.52 kB | ✓ |
| head-HASH.js gzip | 2.28 kB | 2.28 kB | ✓ |
| hooks-HASH.js gzip | 903 B | 903 B | ✓ |
| image-HASH.js gzip | 5.63 kB | 5.63 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 319 B | 319 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 19.1 kB | 19.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| _buildManifest.js gzip | 489 B | 489 B | ✓ |
| Overall change | 489 B | 489 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| index.html gzip | 530 B | 530 B | ✓ |
| link.html gzip | 544 B | 544 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| buildDuration | 13.3s | 12.7s | -576ms |
| buildDurationCached | 5s | 5.1s | ⚠️ +9ms |
| nodeModulesSize | 49.5 MB | 49.5 MB | ⚠️ +1.45 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.05 | 2.98 | -0.07 |
| / avg req/sec | 819.65 | 838.94 | +19.29 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.02 | 2.001 | -0.02 |
| /error-in-render avg req/sec | 1237.81 | 1249.58 | +11.77 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| 17.HASH.js gzip | 2.98 kB | 2.98 kB | ✓ |
| 18.HASH.js gzip | 185 B | 185 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 8.4 kB | 8.4 kB | ✓ |
| webpack-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| Overall change | 68.5 kB | 68.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| _app-HASH.js gzip | 791 B | 791 B | ✓ |
| _error-HASH.js gzip | 3.76 kB | 3.76 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.71 kB | 2.71 kB | ✓ |
| head-HASH.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 3.02 kB | 3.02 kB | ✓ |
| withRouter-HASH.js gzip | 294 B | 294 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 17.6 kB | 17.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| _buildManifest.js gzip | 500 B | 500 B | ✓ |
| Overall change | 500 B | 500 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| index.html gzip | 577 B | 577 B | ✓ |
| link.html gzip | 589 B | 589 B | ✓ |
| withRouter.html gzip | 569 B | 569 B | ✓ |
| Overall change | 1.74 kB | 1.74 kB | ✓ |

</details>

</details>
Commit: 15701f2a702cc2c45ed1bbd614d83b038758fb23

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| buildDuration | 15.3s | 15.5s | ⚠️ +180ms |
| buildDurationCached | 3.6s | 3.4s | -186ms |
| nodeModulesSize | 49.5 MB | 49.5 MB | ⚠️ +1.46 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.731 | 2.547 | -0.18 |
| / avg req/sec | 915.51 | 981.66 | +66.15 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.372 | 1.486 | ⚠️ +0.11 |
| /error-in-render avg req/sec | 1821.62 | 1682.37 | ⚠️ -139.25 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| 359.HASH.js gzip | 2.96 kB | 2.96 kB | ✓ |
| 745.HASH.js gzip | 180 B | 180 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 21 kB | 21 kB | ✓ |
| webpack-HASH.js gzip | 1.53 kB | 1.53 kB | ✓ |
| Overall change | 67.9 kB | 67.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| _app-HASH.js gzip | 803 B | 803 B | ✓ |
| _error-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| amp-HASH.js gzip | 554 B | 554 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.52 kB | 2.52 kB | ✓ |
| head-HASH.js gzip | 2.28 kB | 2.28 kB | ✓ |
| hooks-HASH.js gzip | 903 B | 903 B | ✓ |
| image-HASH.js gzip | 5.63 kB | 5.63 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 319 B | 319 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 19.1 kB | 19.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| _buildManifest.js gzip | 489 B | 489 B | ✓ |
| Overall change | 489 B | 489 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| index.html gzip | 530 B | 530 B | ✓ |
| link.html gzip | 544 B | 544 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| buildDuration | 12.3s | 12s | -218ms |
| buildDurationCached | 4.7s | 4.8s | ⚠️ +96ms |
| nodeModulesSize | 49.5 MB | 49.5 MB | ⚠️ +1.46 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.64 | 2.522 | -0.12 |
| / avg req/sec | 947.03 | 991.4 | +44.37 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.474 | 1.391 | -0.08 |
| /error-in-render avg req/sec | 1696.55 | 1797.11 | +100.56 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| 17.HASH.js gzip | 2.98 kB | 2.98 kB | ✓ |
| 18.HASH.js gzip | 185 B | 185 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 8.4 kB | 8.4 kB | ✓ |
| webpack-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| Overall change | 68.5 kB | 68.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| _app-HASH.js gzip | 791 B | 791 B | ✓ |
| _error-HASH.js gzip | 3.76 kB | 3.76 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.71 kB | 2.71 kB | ✓ |
| head-HASH.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 3.02 kB | 3.02 kB | ✓ |
| withRouter-HASH.js gzip | 294 B | 294 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 17.6 kB | 17.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| _buildManifest.js gzip | 500 B | 500 B | ✓ |
| Overall change | 500 B | 500 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js fix/incorrect-locale-domains | Change |
| - | - | - | - |
| index.html gzip | 577 B | 577 B | ✓ |
| link.html gzip | 589 B | 589 B | ✓ |
| withRouter.html gzip | 570 B | 570 B | ✓ |
| Overall change | 1.74 kB | 1.74 kB | ✓ |

</details>

</details>
Commit: 02be6fcf961fcf8a6db7de01474897bdda96c4e5

**Pull request merged**